### PR TITLE
G.analyse.perday last day qqq window

### PR DESCRIPTION
**Pull request by jhmigueles**
In the last qqq window in the g.analyse.perday function, qqq1 is higher than qqq2, which makes no sense. In this case, the function does not run an error or warning message (even with do.parallel = F), but the file is just not saved into the ms2.out directory.

I set a conditional such as: if (qqq1 > qqq2) next().

It works ok with my files.

### CODE REVIEW

**Comment by jhmigueles** [comment]
It was due to a wrong identification of midnights. We already solved this in g.part5, I made the code consistent here.

This solves the scenario when the dayborder is different to 0am, and the first defined midnight (= dayborder) occurs before the first identified midnight (= 0 am). In this case, the qqq windows were shifted so that the first defined midnight was ignored resulting in a first day with >24h. This error was carried forward till the last qqq window, when the qqq1 was higher than qqq2, and then the function stopped and the files did not get stored in the ms2.out folder.

Now all day windows make sense (24 h except for the first and last depending on the data collection protocol) and the files get appropriately stored.

File: R/g.analyse.R [hunk 1]
```diff
@@ -118,8 +118,21 @@ g.analyse =  function(I,C,M,IMP,qlevels=c(),qwindow=c(0,24),quantiletype = 7,L5M
...
+    }
+    sec = tempp$sec
+    min = tempp$min
+    hour = tempp$hour
+    midnightsi = which(sec == 0 & min == (dayborder-floor(dayborder))*60 & hour == floor(dayborder)) #shift the definition of midnight if required
+    firstmidnight = time[midnightsi[1]]
+    firstmidnighti = midnightsi[1]
+    lastmidnight = time[max(midnightsi)]
+    lastmidnighti = max(midnightsi)
+    midnights = time[midnightsi]
```
[the remark this answers is not in the document]
Comment (jhmigueles) [reply R/g.analyse.R]: Thanks for your feedback.

I did it as you suggested, much better this way. Tested in some files and it works as expected.

Best,
Jairo

**Comment by vincentvanhees** [comment]
Hi Jairo, 
The travis-ci tests are failing for the edits you made. The [report](https://travis-ci.org/wadpac/GGIR/jobs/657239604?utm_medium=notification&utm_source=github_status) says 
 "argument "dayborder" is missing, with no default"
which indicates that we are not providing dayborder as explicit argument to the function R/g.detecmidnight.R. Can you fix this?
Thanks, Vincent

**Comment by jhmigueles** [comment]
Hi Vincent,

Thanks for your help, I planned to look tomorrow into why the tests were failing.

Actually, I included "dayborder" in the g.detecmidnight function arguments:
`g.detecmidnight = function(time,desiredtz,dayborder) {
  convert2clock = function(x) { #ISO format
    return(as.character(format(as.POSIXlt(x,format="%Y-%m-%dT%H:%M:%S%z",tz=desiredtz),"%H:%M:%S")))
  }`

And it is also stated in the function call in g.analyse.R:
`dmidn = g.detecmidnight(time,desiredtz,dayborder)`

Any other idea or am I getting anything wrong? Anyway, if you don't have time for this I will look into it tomorrow as I said. Thanks again!

Best,
Jairo

**Pull request closed without merging**